### PR TITLE
#220 Add explanation/instructions to empty tool window

### LINKED ISSUE #220: Add explanation/instructions to empty tool window
See https://github.com/OpenLiberty/liberty-tools-eclipse/issues/294

### PR DESCRIPTION
**Pull request by Rahul-Biju**
Fixes #220 Added explanation/instructions to empty tool window.

### CODE REVIEW

File: src/main/resources/messages/LibertyBundles.properties [hunk 1]
```diff
@@ -39,7 +39,7 @@ action.io.openliberty.tools.intellij.actions.RemoveLibertyProjectAction.text=Lib
 action.io.openliberty.tools.intellij.actions.RemoveLibertyProjectAction.description=Remove the selected Liberty project from the tool window
 
 # Messages for Liberty actions and pop-up dialog
-no.liberty.projects.detected=No Liberty Maven or Liberty Gradle projects detected in this workspace.
+no.liberty.projects.detected=\n 1. If no projects are open in the Intellij Explorer, open or create a Liberty project using the File menu. \n 2. If there is one or more existing Maven or Gradle projects open in the Intellij Explorer, try one of the following: \n \n \t- Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n \t- Add a server.xml file to an existing Maven or Gradle project at 'src/main/liberty/config'.\n \t- Manually add an existing Maven or Gradle project to the Liberty Dashboard through the Intellij Explorer command palette action 'Liberty:'Add project to Liberty Dashboard'
```
Comment (TrevCraw) [inline src/main/resources/messages/LibertyBundles.properties:42]:
```suggestion
no.liberty.projects.detected=\n 1. If no projects are open in the Project Tool Window, open or create a Liberty project using the File menu. \n 2. If there is one or more existing Maven or Gradle projects open in the Project Tool Window, try one of the following: \n \n \t- Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n \t- Add a server.xml file to an existing Maven or Gradle project at 'src/main/liberty/config'.\n \t- Manually add an existing Maven or Gradle project to the Liberty Tool Window using the 'Liberty: Add project to the tool window' action through the Search Everywhere window.
```

Comment (TrevCraw) [reply src/main/resources/messages/LibertyBundles.properties:42]: Please also make sure to update the spacing based on what was discussed during the demo. I believe the tabs should be removed and there should be an extra new line between (1) and (2).

**Review by TrevCraw — CHANGES_REQUESTED** [review]

**Review by aparnamichael — APPROVED** [review]

File: src/main/resources/messages/LibertyBundles.properties (hunk 1 above)
Comment (Rahul-Biju) [reply src/main/resources/messages/LibertyBundles.properties]: Screenshot 2023-11-24 at 7 43 53 PM

Addressed the above review comment.

Comment (TrevCraw) [reply src/main/resources/messages/LibertyBundles.properties]: @Rahul-Biju you addressed the spacing issue, but you did not change the wording from the first review comment: https://github.com/OpenLiberty/liberty-tools-intellij/pull/584#discussion_r1403932384

Comment (Rahul-Biju) [reply src/main/resources/messages/LibertyBundles.properties]: @TrevCraw
 Word changes : Done.

Comment (TrevCraw) [reply src/main/resources/messages/LibertyBundles.properties]: @Rahul-Biju The updates to the message look good, but you did not update the message in the corresponding test case

File: src/main/resources/messages/LibertyBundles.properties [hunk 2]
```diff
@@ -34,12 +34,12 @@ action.io.openliberty.tools.intellij.actions.RefreshLibertyToolbar.description=R
 action.io.openliberty.tools.intellij.actions.RunLibertyDevTask.text=Liberty: Run Liberty action
 action.io.openliberty.tools.intellij.actions.RunLibertyDevTask.description=Run the selected Liberty action
 action.io.openliberty.tools.intellij.actions.AddLibertyProjectAction.text=Liberty: Add project to the tool window
-action.io.openliberty.tools.intellij.actions.AddLibertyProjectAction.description=Add the selected Liberty project to the tool window
+action.io.openliberty.tools.intellij.actions.AddLibertyProjectAction.description=Add the selected Liberty project to the tool window
```
Comment (TrevCraw) [inline src/main/resources/messages/LibertyBundles.properties:37]: please remove the extra space at the end - there is no need to have any changes on this line

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 3]
```diff
@@ -1691,7 +1692,13 @@ public static void selectProjectFromRemoveLibertyProjectDialog(RemoteRobot remot
...
      */
     public static void waitForLTWNoProjectDetectedMsg(RemoteRobot remoteRobot, int waitTime) {
-        String text = "No Liberty Maven or Liberty Gradle projects detected in this workspace.";
+        String text = " 1. If no projects are open in the Intellij Explorer, open or create a Liberty project using the" +
+                " File menu.  2. If there is one or more existing Maven or Gradle projects open in the Intellij " +
+                "Explorer, try one of the following:   a. Configure the Liberty build plugin in the build file of an " +
+                "existing Maven or Gradle project. b. Add a server.xml file to an existing Maven or Gradle project" +
+                " at 'src/main/liberty/config'. c. Manually add an existing Maven or Gradle project to the Liberty " +
+                "Dashboard through the Intellij Explorer command palette action 'Liberty:'Add project to Liberty " +
+                "Dashboard'";
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:1701]: please update the test case message to match

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: src/main/resources/messages/LibertyBundles.properties [hunk 4]
```diff
@@ -34,12 +34,12 @@ action.io.openliberty.tools.intellij.actions.RefreshLibertyToolbar.description=R
...
 action.io.openliberty.tools.intellij.actions.RemoveLibertyProjectAction.text=Liberty: Remove project from the tool window
 action.io.openliberty.tools.intellij.actions.RemoveLibertyProjectAction.description=Remove the selected Liberty project from the tool window
 
 # Messages for Liberty actions and pop-up dialog
-no.liberty.projects.detected=No Liberty Maven or Liberty Gradle projects detected in this workspace.
+no.liberty.projects.detected=\n 1. If no projects are open in the Project Tool Window, open or create a Liberty project using the File menu. \n\n 2. If there is one or more existing Maven or Gradle projects open in the Project Tool Window, try one of the following: \n \n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n\n b. Add a server.xml file to an existing Maven or Gradle project at 'src/main/liberty/config'.\n\n c. Manually add an existing Maven or Gradle project to the Liberty Tool Window using the 'Liberty: Add project to the tool window' action through the Search Everywhere window.
```
Comment (dmuelle) [inline src/main/resources/messages/LibertyBundles.properties:42]:
```suggestion
no.liberty.projects.detected=\n 1. If no projects are open in the Project Tool Window, open or create a Liberty project using the File menu. \n\n 2. If one or more existing Maven or Gradle projects are open in the Project Tool Window, try one of the following actions: \n \n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n\n b. Add a server.xml file to an existing Maven or Gradle project at 'src/main/liberty/config'.\n\n c. Manually add an existing Maven or Gradle project to the Liberty Tool Window using the 'Liberty: Add project to the tool window' action through the Search Everywhere window.
```

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 5]
```diff
@@ -1691,7 +1692,13 @@ public static void selectProjectFromRemoveLibertyProjectDialog(RemoteRobot remot
      * @param waitTime    The time (seconds) to wait for the required message to appear in the text area.
      */
     public static void waitForLTWNoProjectDetectedMsg(RemoteRobot remoteRobot, int waitTime) {
-        String text = "No Liberty Maven or Liberty Gradle projects detected in this workspace.";
+        String text = " 1. If no projects are open in the Intellij Explorer, open or create a Liberty project using the" +
+                " File menu.  2. If there is one or more existing Maven or Gradle projects open in the Intellij " +
```
Comment (dmuelle) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:1696]:
```suggestion
                " File menu.  2. If one or more existing Maven or Gradle projects are open in the Intellij " +
```

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 6]
```diff
@@ -1691,7 +1692,13 @@ public static void selectProjectFromRemoveLibertyProjectDialog(RemoteRobot remot
...
+                "Explorer, try one of the following:   a. Configure the Liberty build plugin in the build file of an " +
```
Comment (dmuelle) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:1697]:
```suggestion
                "Explorer, try one of the following actions:   a. Configure the Liberty build plugin in the build file of an " +
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

**Comment by TrevCraw** [comment]
@Rahul-Biju please incorporate the changes suggested by @dmuelle and myself

File: src/main/resources/messages/LibertyBundles.properties (hunk 2 above)
Comment (Rahul-Biju) [reply src/main/resources/messages/LibertyBundles.properties]: Changes : done

**Review by dmuelle — DISMISSED** [review]
LGTM

File: src/main/resources/messages/LibertyBundles.properties [hunk 7]
```diff
@@ -39,7 +39,7 @@ action.io.openliberty.tools.intellij.actions.RemoveLibertyProjectAction.text=Lib
 action.io.openliberty.tools.intellij.actions.RemoveLibertyProjectAction.description=Remove the selected Liberty project from the tool window
 
 # Messages for Liberty actions and pop-up dialog
-no.liberty.projects.detected=No Liberty Maven or Liberty Gradle projects detected in this workspace.
+no.liberty.projects.detected=\n 1. If no projects are open in the Project Tool Window, open or create a Liberty project using the File menu. \n\n 2. If one or more existing Maven or Gradle projects are open in the Project Tool Window, try one of the following actions: \n \n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n\n b. Add a server.xml file to an existing Maven or Gradle project at 'src/main/liberty/config'.\n\n c. Manually add an existing Maven or Gradle project to the Liberty Tool Window using the 'Liberty: Add project to the tool window' action through the Search Everywhere window.
```
Comment (yeekangc) [inline src/main/resources/messages/LibertyBundles.properties:42]:
```suggestion
no.liberty.projects.detected=\n 1. If no projects are open in the Project tool window, open or create a Liberty project using the File menu. \n\n 2. If one or more existing Maven or Gradle projects are open in the Project tool window, try one of the following actions: \n \n a. Configure the Liberty build plugin in the build file of an existing Maven or Gradle project.\n\n b. Add a server.xml file to an existing Maven or Gradle project at 'src/main/liberty/config'.\n\n c. Manually add an existing Maven or Gradle project to the Liberty tool window using the 'Liberty: Add project to the tool window' action through the Search Everywhere window.
```

**Review by yeekangc — CHANGES_REQUESTED** [review]

**Comment by TrevCraw** [comment]
@Rahul-Biju the test case message still does not match the message we are now displaying in the Liberty tool window. Please update the test case and ensure the builds pass.

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java (hunk 3 above)
Comment (dmuelle) [reply src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:1701]: should the text here match what is in LibertyBundles.properties? It looks ok on its own but im not sure based on your comment if it's still awaiting changes to be like the other file

Comment (TrevCraw) [reply src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:1701]: @dmuelle Yes, the text should be updated here as well. If you are good with the text as it is in LibertyBundles.properties, feel free to approve. No need to wait for the other changes to come in, they are just for testing. I will make sure that gets handled.

**Review by dmuelle — DISMISSED** [review]
LGTM, thanks

**Review by yeekangc — APPROVED** [review]

**Review by aparnamichael — DISMISSED** [review]
Looks good to me

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java [hunk 8]
```diff
@@ -19,6 +19,7 @@
 import io.openliberty.tools.intellij.it.fixtures.DialogFixture;
 import io.openliberty.tools.intellij.it.fixtures.ProjectFrameFixture;
 import io.openliberty.tools.intellij.it.fixtures.WelcomeFrameFixture;
+import io.openliberty.tools.intellij.util.LocalizedResourceUtil;
```
Comment (TrevCraw) [inline src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java:22]: this import is not being used, please remove

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java (hunk 8 above)
Comment (Rahul-Biju) [reply src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java]: @TrevCraw removed import.

**Review by aparnamichael — APPROVED** [review]

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**